components/datetime/src/provider/mod.rs

```diff
@@ -193,5 +194,91 @@ pub mod gregory {
+            )]
+            pub count: Option<CountV1>,
+        }
+
+        // Breakdown of pattern utf8 byte sizes for all of CLDR data, as of 2020-02.
+        //
+        // See: https://github.com/unicode-org/icu4x/pull/480#discussion_r573028611
+        //
+        // Byte range | Pattern Count | Pattern %
+        // -------------------------------------------------
+        //  1 - 10             22,526      79.61%
+        // 11 - 20              4,784      16.91%
+        // 21 - 30                841       2.97%
+        // 31 - 40                 69       0.24%
+        // 41 - 50                 75       0.27%
+
+        // Patterns with variants, as of 2020-02.
+        //
+        // Variant count | Skeleton count | Percentage | Percentage Cumulative
+        // -------------------------------------------------------------------
+        //             1           26362         97.1%                   97.1%
+        //             2             658          2.4%                   99.5%
+        //             3              37          0.1%                   99.6%
+        //             4              32          0.1%                   99.8%
+        //             5               6          0.0%                   99.8%
+        //             6              58          0.2%                  100.0%
+        //
+        // Choosing a SmallVec length of 1 covers 97.1% of all cases.
+
+        // Skeleton sizes, as of 2020-02. Note that skeletons are valid ascii characters, and
+        // thus are only 1 byte each character. Patterns do not have this guarantee.
+        //
+        // Character count | Total Skeletons | Percentage | Percentage Cumulative
+        // ======================================================================
+        //               1              3390        12.0%                   12.0%
+        //               2              5655        20.0%                   32.0%
+        //               3              5892        20.8%                   52.8%
+        //               4              5948        21.0%                   73.8%
+        //               5              5146        18.2%                   92.0%
+        //               6              1470         5.2%                   97.2%
+        //               7               727         2.6%                   99.8%
+        //               8                43         0.2%                   99.9%
+        //               9                16         0.1%                  100.0%
+        //              10                 8         0.0%                  100.0%
+
+        // TODO - This could have better memory locality with TinyStr, however it does not
+        // currently implement Serialize or Deserialize traits.
+        //
+        // pub type SkeletonsV1 = Vec<(TinyStr16, SmallVec<[SkeletonPatternV1; 1]>)>,
+        pub type SkeletonsV1 = Vec<(Cow<'static, str>, SmallVec<[SkeletonPatternV1; 1]>)>;
```

Comment:
Suggestion: instead of using `SmallVec<[SkeletonPatternV1; 1]>`, consider an `enum` with two variants: a single pattern, and a TupleVecMap of multiple patterns keyed by plural form.